src/components/calcite-modal/calcite-modal.tsx

```diff
@@ -170,7 +156,12 @@ export class CalciteModal {
         if (this.firstFocus) {
           this.firstFocus.focus();
         } else {
-          this.trap.focusFirstElement();
+          const focusableElements = queryShadowRoot(this.el, isHidden, isFocusable);
+          if (focusableElements.length > 0) {
```

Comment:
Maybe here you could call `this.focusFirstElement`? It seems like that's what you're up to here.

Also, it seems like this is doing more work than `this.focusFirstElement`. That method only does `this.closeButton.focus()`, but here you're getting a list of focussable elements and focussing the first one.